cmd/dlt/ocmrole/cmd.go

```diff
@@ -0,0 +1,239 @@
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVar(
+		&args.roleARN,
+		"role-arn",
+		"",
+		"Role ARN to delete from the OCM organization account")
+
+	aws.AddModeFlag(Cmd)
+
+	confirm.AddFlag(flags)
+	interactive.AddFlag(flags)
+}
+
+func run(cmd *cobra.Command, argv []string) error {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+	awsClient := aws.CreateNewClientOrExit(logger, reporter)
+	ocmClient := ocm.CreateNewClientOrExit(logger, reporter)
+	defer func() {
+		err := ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	mode, err := aws.GetMode()
+	if err != nil {
+		reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
+	orgID, _, err := ocmClient.GetCurrentOrganization()
+	if err != nil {
+		reporter.Errorf("Error getting organization account: %v", err)
+		return err
+	}
+
+	if len(argv) > 0 {
+		args.roleARN = argv[0]
+	}
+
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && (!cmd.Flags().Changed("mode")) {
+		interactive.Enable()
+	}
+
+	if reporter.IsTerminal() {
+		reporter.Infof("Deleting OCM role")
+	}
+
+	roleARN := args.roleARN
+
+	if !interactive.Enabled() && roleARN == "" {
+		interactive.Enable()
+	}
+
+	if interactive.Enabled() {
+		roleARN, err = interactive.GetString(interactive.Input{
+			Question: "OCM Role ARN",
+			Help:     cmd.Flags().Lookup("role-arn").Usage,
+			Default:  roleARN,
+			Required: true,
+			Validators: []interactive.Validator{
+				aws.ARNValidator,
+			},
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid ocm role ARN to delete from the current organization: %s", err)
+			os.Exit(1)
+		}
+	}
+	if roleARN != "" {
+		_, err := arn.Parse(roleARN)
+		if err != nil {
+			reporter.Errorf("Expected a valid ocm role ARN to delete from the current organization: %s", err)
+			os.Exit(1)
+		}
+	}
+	if !confirm.Prompt(true, "Delete the '%s' role from organization '%s'?", roleARN, orgID) {
+		os.Exit(0)
+	}
+
+	linkedRoles, err := ocmClient.GetOrganizationLinkedOCMRoles(orgID)
+	if err != nil {
+		reporter.Errorf("An error occurred while trying to get the organization linked roles: %s", err)
+		os.Exit(1)
+	}
+	isLinked := helper.Contains(linkedRoles, roleARN)
+
+	if interactive.Enabled() {
+		mode, err = interactive.GetOption(interactive.Input{
+			Question: "OCM role deletion mode",
+			Help:     cmd.Flags().Lookup("mode").Usage,
+			Default:  aws.ModeAuto,
+			Options:  aws.Modes,
+			Required: true,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid OCM role deletion mode: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	roleName, err := aws.RoleARNToRoleName(roleARN)
+	if err != nil {
+		reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+	switch mode {
+	case aws.ModeAuto:
+		ocmClient.LogEvent("ROSADeleteOCMRoleModeAuto", nil)
+		if isLinked {
+			reporter.Warnf("Role ARN '%s' is linked to organization '%s'. Unlink the role before deletion", roleARN, orgID)
```

Comment:
Either remove Unlink the role before deletion or change it to Unlinking the role before deletion",

Reply:
Removed.
I've made a small change in `unlink/ocmrole/cmd.go` so now we ask for role ARN only once (updated the screenshot in the comment).
Thanks.

Reply:
Though now the output of the `rosa unlink ocm-role` is a bit different:
![image](https://user-images.githubusercontent.com/57869309/154048818-9c20aab0-ac36-4ee4-9a89-5ce976c879e1.png)
In case the `roleARN` is provided, it won't prompt:
`? OCM Role ARN: arn:aws:iam::765374464689:role/ManagedOpenShift-OCM-Role-12541229`
WDYT?

Alternatively, I can pass a boolean argument from `rosa delete ocm-role` to determine if we should ask for `roleARN`.
Or there is a way to determine from with `rosa unlink ocm-role` if it was called by `rosa delete ocm-role`?